sdcflows/interfaces/fmap.py

```diff
@@ -661,19 +661,31 @@ def _delta_te(in_values, te1=None, te2=None):
 
 def au2rads(in_file, newpath=None):
     """Convert the input phase difference map in arbitrary units (a.u.) to rads."""
-    from scipy.stats import mode
+    from scipy import stats
     im = nb.load(in_file)
     data = im.get_fdata(dtype='float32')
     hdr = im.header.copy()
 
-    # First center data around 0.0.
-    data -= mode(data, axis=None)[0][0]
+    dmin, dmax = data.min(), data.max()
+
+    # Find the mode ignoring outliers on the far max/min, to allow for junk outside the FoV
+    fudge = 0.05 * (dmax - dmin)
+    mode = stats.mode(data[(data > dmin + fudge) & (data < dmax - fudge)])[0][0]
```

Comment:
We still want to compute over the whole array right?

```suggestion
    mode = stats.mode(data[(data > dmin + fudge) & (data < dmax - fudge)], axis=None)[0][0]
```

Reply:
Boolean indexing flattens the array.